components/feature/qr/src/main/java/mozilla/components/feature/qr/CustomOverlay.kt

```diff
@@ -0,0 +1,105 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.qr
+
+import android.app.Activity
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.Canvas
+import android.graphics.Color
+import android.graphics.Paint
+import android.graphics.PorterDuff
+import android.graphics.PorterDuffXfermode
+import android.graphics.RectF
+import android.util.AttributeSet
+import android.util.DisplayMetrics
+import android.widget.ImageView
+import android.widget.LinearLayout
+
+/**
+ * A custom overlay [View] that sets the opacity of the background to dim it.
+ * This gives a focus like look to the qr reticle.
+ * @property windowFrame a reference to a [Bitmap], used to create the window frame.
+ * @property activity a reference to an [Activity], used to refer the child views of
+ * calling parent view.
+ * @property ALPHA defines the opacity of the background.
+ * @property RADIUS defines the radius of the rectangle corners.
+ */
+
+class CustomOverlay : LinearLayout {
+    private var windowFrame: Bitmap? = null
+    private lateinit var activity: Activity
```

Comment:
I think you can just directly set this:
```suggestion
    private val activity = context as Activity
```